contracts/LotteryNFT.sol

```diff
@@ -1,38 +1,94 @@
-pragma solidity 0.6.12;
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.8.0;
 pragma experimental ABIEncoderV2;
 
-import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "@nomiclabs/buidler/console.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+
 
-contract LotteryNFT is ERC721, Ownable {
+contract LotteryNFT is ERC1155, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
+    
+    uint256
+        private constant LOTTERY_INDEX = 0x00000000000000000000000000000000000000000000000000000000000007FF;
+    
+    address payable public manager;
+    string public name = "Pancake Lottery Ticket";
+    string public symbol = "PLT";
 
     mapping (uint256 => uint8[4]) public lotteryInfo;
     mapping (uint256 => uint256) public lotteryAmount;
     mapping (uint256 => uint256) public issueIndex;
     mapping (uint256 => bool) public claimInfo;
 
-    constructor() public ERC721("Pancake Lottery Ticket", "PLT") {}
+
+    event Ticket(
+        address indexed _player,
+        uint256 indexed _id,
+        uint256 indexed _amount
+    );
+
+    event TicketBatch(
+        address indexed _player,
+        uint256[] indexed _ids,
+        uint256[] indexed _amounts
+    );
+
+    modifier onlyManager() {
+        require(msg.sender == manager, "caller is not the manager");
+        _;
+    }
+
+    constructor(
+        string memory _newURI,
+        address payable _deployerAddress
+    ) public ERC1155(_newURI) {
+        name;
+        symbol;
+        manager = _deployerAddress;
+    }
 
     function newLotteryItem(address player, uint8[4] memory _lotteryNumbers, uint256 _amount, uint256 _issueIndex)
-        public onlyOwner
+        public onlyManager()
         returns (uint256)
     {
         _tokenIds.increment();
-
         uint256 newItemId = _tokenIds.current();
-        _mint(player, newItemId);
+        _mint(player, newItemId, _amount, "");
         lotteryInfo[newItemId] = _lotteryNumbers;
         lotteryAmount[newItemId] = _amount;
         issueIndex[newItemId] = _issueIndex;
         // claimInfo[newItemId] = false; default is false here
         // _setTokenURI(newItemId, tokenURI);
-
+        
+        emit Ticket(player, newItemId, _amount);
         return newItemId;
+        
     }
 
+    function newBatchLotteryItem(uint256[] memory _ids, address player, uint8[4] memory _lotteryNumbers, uint256[] memory _amounts, uint256[] memory _issueIndexs)
```

Comment:
where will the lottery contract call it?

Reply:
I will add call.